File: src/worker/java/MySIWorker.java
import parspice.sender.IntSender;
import parspice.sender.LongSender;
import parspice.worker.SIWorker;

public class MySIWorker extends SIWorker<Long, Integer> {

    public MySIWorker() {
        super(new LongSender(), new IntSender());
    }

    @Override
    public void setup(Long input) throws Exception {
        // do something with input
    }

    @Override
    public void task(Integer input) throws Exception {
        // do something with input
    }
}

File: src/worker/kotlin/MySIWorker.kt
import parspice.sender.IntSender
import parspice.sender.LongSender
import parspice.worker.SIWorker

class MySIWorker: SIWorker<Long, Int>(LongSender(), IntSender()) {
    override fun setup(input: Long) {}

    override fun task(i: Int) {}
}